### PR TITLE
Dt 1772 amend contact

### PR DESCRIPTION
**Pull request by andymarke**
* Can navigate to the amend contacts page
* TODO - validate form
* TODO - save and update contacts

### CODE REVIEW

File: server/routes/courtRegister/courtRegisterController.test.ts [hunk 1]
```diff
@@ -640,47 +656,105 @@ describe('Court Register controller', () => {
         })
       })
     })
-  })
-  describe('parseFilter', () => {
```
Comment (andymarke) [inline server/routes/courtRegister/courtRegisterController.test.ts:644]: this was accidentally duplicated, so removed

File: server/routes/courtRegister/courtRegisterController.ts [hunk 2]
```diff
@@ -306,4 +307,36 @@ export default class CourtRegisterController {
...
+  }
+
+  async amendCourtBuildingContacts(req: Request, res: Response): Promise<void> {
+    const view = new AmendCourtBuildingContactsView(req.session.amendCourtBuildingContactsForm, req.flash('errors'))
+
+    res.render('pages/court-register/amendCourtBuildingContacts', view.renderArgs)
+  }
+
+  async submitAmendCourtBuildingContacts(req: Request, res: Response): Promise<void> {
+    req.session.amendCourtBuildingContactsForm = { ...req.body }
```
Comment (andymarke) [inline server/routes/courtRegister/courtRegisterController.ts:339]: TODO - save data and redirect back to court details

File: server/views/pages/court-register/amendCourtBuildingContacts.njk [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    <!--suppress HtmlUnknownTarget -->
+    <main class="app-container govuk-body">
+        {% if errors.length > 0 %}
+            {{ govukErrorSummary({
+                titleText: 'There is a problem',
+                errorList: errors,
+                attributes: { 'data-qa-errors': true }
+            }) }}
+        {% endif %}
+        <div class="govuk-grid-row" data-module="moj-add-another">
```
Comment (andymarke) [inline server/views/pages/court-register/amendCourtBuildingContacts.njk:22]: using MOJ "add another" component

File: integration_tests/integration/court-register-amend.spec.ts [hunk 4]
```diff
@@ -187,4 +188,33 @@ context('Court register - amend existing court', () => {
...
+      courtDetailsPage.amendBuildingContactsLink('2').should('contain.text', 'Change')
+    })
+    it('will display existing contact details with ability to remove a number', () => {
+      const courtDetailsPage = CourtDetailsPage.verifyOnPage('Sheffield Magistrates Court')
+      courtDetailsPage.amendBuildingContactsLink('1').click()
+      const amendCourtBuildingContactsPage = AmendCourtBuildingContactsPage.verifyOnPage('Sheffield Courts')
+
+      amendCourtBuildingContactsPage.number(0).should('have.value', '0114 555 1234')
+      amendCourtBuildingContactsPage.type(0).should('have.value', 'TEL')
+      amendCourtBuildingContactsPage.removeButton(0)
```
Comment (mikehalmamoj) [inline integration_tests/integration/court-register-amend.spec.ts:213]: Should we assert on the text in the remove button?

Comment (andymarke) [reply integration_tests/integration/court-register-amend.spec.ts:213]: I was going to say no, but actually since we get by id that would be good. Next PR

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**